### PR TITLE
Easy and safe database with macro

### PR DESCRIPTION
**Pull request by namse**
Breaking changes
- All partition's prefix become the name of document struct.

The goal of this PR
- Making sure which parameter is partition key and sort key. 
  - Previous function only get anonymous parameter so user can put wrong data. For example, if the parameter type is String, user can put `user.id` or `project.id`. i don't like it, so i tried to make it sure.

# Before & After

```rust

// Before

#[derive(Clone, serde::Serialize, serde::Deserialize)]
pub struct UserDocument {
    pub id: rpc::Uuid,
}

impl Document for UserDocument {
    fn partition_key_prefix() -> &'static str {
        "user"
    }

    fn partition_key_without_prefix(&self) -> String {
        self.id.to_string()
    }

    fn sort_key(&self) -> Option<String> {
        None
    }
}

// After

#[document_macro::document]
pub struct UserDocument {
    #[pk]
    pub id: rpc::Uuid,
}
```

```rust

// Before

let user_document = crate::dynamo_db()
    .get_item::<UserDocument>(&identity_document.user_id, Option::<String>::None)
    .await
    .map_err(|error| GetOrCreateUserError::GetUserError(error))?;



// After
            
let user_document = UserDocumentGet {
    pk_id: identity_document.user_id,
}
.run()
.await
.map_err(|error| GetOrCreateUserError::GetUserError(error))?;

### CODE REVIEW

**Review by bigfoodK — APPROVED** [review]

File: luda-editor/server/src/services/project/documents.rs [hunk 1]
```diff
@@ -1,85 +1,112 @@
...
+//     > {
+//         crate::storage::dynamo_db::TransactUpdateCommand {
+//             partition_prefix: "project_acl_user_in".to_string(),
+//             partition_key_without_prefix: self.pk_user_id.to_string(),
+//             sort_key: Some(self.sk_project_id.to_string()),
+//             update: self.update,
+//             _phantom: std::marker::PhantomData,
+//         }
+//     }
+// }
```
[the remark this answers is not in the document]
Comment (namse) [reply luda-editor/server/src/services/project/documents.rs:112]: naver

**Comment by bigfoodK** [comment]
I have a question.
1. `pk`: primary key, `sk`: secondary key
2. `pk`: partition key, `sk`: ???
Is there an answer between the two?

**Comment by namse** [comment]
> 

`p`artition `k`ey and `s`ort `k`ey of dynamodb

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html#HowItWorks.CoreComponents.PrimaryKey

**Comment by bigfoodK** [comment]
Thank you

**Pull request merged**